### PR TITLE
integrate core components into odyssey

### PR DESCRIPTION
**Pull request by ganeshsomasundaram-okta**
Starter storybook playground
image

### CODE REVIEW

File: packages/core-components/package.json
Comment (jmelberg-okta) [inline packages/core-components/package.json:23]: This will need to be a public artifact for us to consume it in an open source project. I wonder if we can hold off on adding this for now...

File: packages/core-components/src/components/button/button.scss
Comment (jmelberg-okta) [inline packages/core-components/src/components/button/button.scss:3]: We shouldn't link directly to the Odyssey source here. Instead, let's import it as we would any CSS project.

File: packages/core-components/src/components/button/button.tsx
Comment (jmelberg-okta) [inline packages/core-components/src/components/button/button.tsx:6]: [Discussed briefly in my stale PR](https://github.com/okta/odyssey/pull/687/files#r496135166) - but we should adhere to the same prefix for all Odyssey components:

```suggestion
@Component({ tag: 'ods-button', styleUrl: 'button.scss', shadow: true })
```

File: packages/core-components/src/components/button/button.tsx
Comment (jmelberg-okta) [inline packages/core-components/src/components/button/button.tsx:12]: Seems like a great opportunity to tackle all these `TODO` items before merging!

File: packages/core-components/src/components/icon/sass/_classes.scss
Comment (jmelberg-okta) [inline packages/core-components/src/components/icon/sass/_classes.scss:131]: We have new requirements when embedding SVGs into our repo (ex: SVGO). Can we move these into the `odyssey-icons` package?

File: packages/core-components/src/types.ts
Comment (jmelberg-okta) [inline packages/core-components/src/types.ts:1]: Just an FYI too - since these are all source files we'll need to include a [legal header](https://github.com/okta/okta-auth-js/blob/master/lib/OktaAuthBase.ts#L1-L11).

File: packages/core-components/src/utils/dom-utils.ts
Comment (jmelberg-okta) [inline packages/core-components/src/utils/dom-utils.ts:4]: IIRC we need to call this out in a `THIRD_PARTY_DEPENDENCIES` file - as we're embedding another person/company's source code.

File: packages/core-components/src/utils/test-utils.ts
Comment (jmelberg-okta) [inline packages/core-components/src/utils/test-utils.ts:27]: We should loop in the Odyssey team to agree on what we want to validate here. These were specific to `admin`.

File: packages/core-components/src/utils/utils.ts
Comment (jmelberg-okta) [inline packages/core-components/src/utils/utils.ts:21]: Are all these functions currently used? If not - lets just remove them.

File: scripts/build-profiles/stencil-component/commands/build.js
Comment (jmelberg-okta) [inline scripts/build-profiles/stencil-component/commands/build.js:8]: Does this mean we'll need a separate package to contain a build profile for each component type? Instead, what if we exposed multiple choices and imported the correct build profile from there?

Example in [this PR](https://github.com/okta/odyssey/blob/f3535e8806d1f91577e656b6dbdfa9954d8a5bac/scripts/tools/commands/build.js).

File: scripts/build-profiles/stencil-component/commands/e2e.js
Comment (jmelberg-okta) [inline scripts/build-profiles/stencil-component/commands/e2e.js:30]: We likely will not use Bacon for this - so we will need to hook into Travis CI to ensure these are reported correctly.

File: scripts/build-profiles/stencil-component/package.json
Comment (jmelberg-okta) [inline scripts/build-profiles/stencil-component/package.json:10]: This is another internal dependency we will not have access to - so let's remove it.

File: scripts/build-profiles/stencil-component/package.json
Comment (jmelberg-okta) [inline scripts/build-profiles/stencil-component/package.json:28]: nit: Let's use up to date versions of these dependencies if we can!

File: yarn.lock
Comment (jmelberg-okta) [inline yarn.lock:7]: Make sure to move your `.npmrc` so we don't pick up our internal artifactory!

**Review by jmelberg-okta — COMMENTED** [review]
Mostly questions!

File: packages/core-components/.storybook/stories/welcome/welcome.css
Comment (arnoldsandoval-okta) [inline packages/core-components/.storybook/stories/welcome/welcome.css:1]: This isn't necessary, we will be importing this type of content from Markdown -- Storybook provides the default style for this kind of documentation.

Please remove the file

Related thought: https://github.com/okta/odyssey/pull/912/files#r560609756

File: packages/core-components/src/components/button/button.e2e.ts
Comment (arnoldsandoval-okta) [inline packages/core-components/src/components/button/button.e2e.ts:7]: All `o-` prefixes should be changed to `ods-` to maintain consistency with the system

File: packages/core-components/src/components/button/button.scss
Comment (arnoldsandoval-okta) [inline packages/core-components/src/components/button/button.scss:9]: This file shouldnt exist, and instead should come straight from Odyssey. If there is a reason for having this CSS _which all consumers care about_ we should integrate it directly into `@okta/odyssey` instead of doing it this way.

Otherwise, this code should live within `@okta/admin-app`

File: packages/core-components/src/components/icon/icon.tsx
Comment (arnoldsandoval-okta) [inline packages/core-components/src/components/icon/icon.tsx:1]: This component should be refactored to use the proper Odyssey icon implementation.

I understand this may be out of scope for this PR. If that is the case, please remove this and any references to Icon!

File: packages/core-components/src/globals/all.scss
Comment (arnoldsandoval-okta) [inline packages/core-components/src/globals/all.scss:1]: okticon is not supported by Odyssey, please remove

File: packages/core-components/src/utils/dom-utils.ts
Comment (arnoldsandoval-okta) [reply packages/core-components/src/utils/dom-utils.ts:4]: Is there any reason why the license/comment isn't at the very top of the file?

File: scripts/build-profiles/stencil-component/package.json
Comment (arnoldsandoval-okta) [inline scripts/build-profiles/stencil-component/package.json:16]:
```suggestion
```

This dep is not necessary as storybook provides a way to do this out of the box

Here is a reference implementation from my branch: https://github.com/okta/odyssey/blob/storybook/packages/storybook/stories/Tooltip.stories.js#L8

File: scripts/build-profiles/stencil-component/package.json
Comment (arnoldsandoval-okta) [inline scripts/build-profiles/stencil-component/package.json:17]:
```suggestion
```

Please remove in favor of storybook controls

File: scripts/build-profiles/stencil-component/package.json
Comment (arnoldsandoval-okta) [inline scripts/build-profiles/stencil-component/package.json:19]:
```suggestion
```

Please remove, functionality now included in storybook core

File: scripts/build-profiles/stencil-component/package.json
Comment (arnoldsandoval-okta) [inline scripts/build-profiles/stencil-component/package.json:27]: I dont see emotion being used anywhere in this PR, we should remove this.

File: packages/core-components/README.md
Comment (arnoldsandoval-okta) [inline packages/core-components/README.md:1]: This is great, I have added it to [OKTA-361478] (https://oktainc.atlassian.net/browse/OKTA-361478)

File: packages/core-components/src/components.d.ts
Comment (arnoldsandoval-okta) [inline packages/core-components/src/components.d.ts:10]: All O prefixes should be Ods

File: packages/core-components/src/components/icon/sass/_classes.scss
Comment (arnoldsandoval-okta) [inline packages/core-components/src/components/icon/sass/_classes.scss:1]: Please remove this file for reasons stated in this PR.

**Review by arnoldsandoval-okta — APPROVED** [review]

**Review by arnoldsandoval-okta — CHANGES_REQUESTED** [review]
Thanks for doing this @ganeshsomasundaram-okta, my position on this is that we should start by introducing only the bare minimum to get something that resembles `core.components` running in the repo.

I think this PR does that, but I'm noticing some (what I could consider) cruft. Mainly things like legacy CSS (and possibly other code/scripts), and unused dependencies. I understand this is a Draft PR, so I'm assuming you're still working through it.

Also, I would love for us to consider a fresh install of Storybook, as quite a lot has changed in the past year. Here is a WIP [reference implementation](https://e8dea06-storybook.ods.dev) of what that could look like.

If there's any way I can help, please let me know!

File: packages/core-components/package.json
Comment (edburyenegren-okta) [inline packages/core-components/package.json:3]: We'll want to make sure than anything versioned matches our current/planned ODS release.

File: packages/core-components/src/utils/test-utils.ts
Comment (edburyenegren-okta) [reply packages/core-components/src/utils/test-utils.ts:27]: Correct! No plans for additional compliance rn.

**Pull request closed without merging**
